cpp/dolfinx/fem/utils.cpp

```diff
@@ -155,6 +155,18 @@ la::PETScMatrix dolfinx::fem::create_matrix(const Form& a)
   pattern.assemble();
```

Comment:
I think this line should be removed and let the caller finalise the sparsity pattern.